### ISSUE TITLE
cfLetAutoLib is sensitive to load order

### ISSUE THREAD

**Issue by oskarabrahamsson** [CakeML/cakeml#699]
`cfLetAutoLib` behaving differently depending on when it is loaded. This issue is about making the code robust against that.

Here is an example: the simpset `DEF_SIMPSET` is used by some of the library internals. This simpset is `srw_ss` taken at some point in time, plus whatever you add to it later using the interface. Depending on when this code is run, `xlet_auto` will know different things. [Here is the relevant code](https://github.com/CakeML/cakeml/blob/dc73bd8f21601e06bdc92eb303865a3992673970/characteristic/cfLetAutoLib.sml#L220):
```sml
val DEF_SIMPSET = ref pure_ss;
val _ = (DEF_SIMPSET := srw_ss());

(* TODO: Find a way to export that - like with ThmSetData.new_exporter *)
fun add_simp_frag sf = (DEF_SIMPSET := ((!DEF_SIMPSET) ++ sf));
fun get_default_simpset () = !DEF_SIMPSET;
```

There may be other things like this.

**Comment by mn200** [comment]
I'd say that DEF_SIMPSET should be built on a fixed base (`list_ss` perhaps) and then augmented by a fixed set of fragments grabbed by using `BasicProvers.thy_ssfrag`.

**Comment by konrad-slind** [comment]
I agree. Things that depend on stateful simpsets are fragile.

**Comment by oskarabrahamsson** [comment]
@mn200 Thanks for the suggestion.

This issue can be closed once #703 is merged.
